src/huggingface_hub/inference/_common.py

```diff
@@ -197,6 +198,18 @@ def _b64_encode(content: ContentT) -> str:
         return base64.b64encode(data_as_bytes).decode()
 
 
+def _as_url(content: ContentT, default_mime_type: str) -> str:
+    if isinstance(content, str) and (content.startswith("https://") or content.startswith("http://")):
+        return content
+
+    mime_type: Optional[str] = None
+    if isinstance(content, (str, Path)):
+        mime_type, _ = mimetypes.guess_type(str(content))
+    final_mime_type = mime_type or default_mime_type
```

Comment:
```suggestion
    mime_type = (
        mimetypes.guess_type(content, strict=True)[0]
        if isinstance(content, (str, Path))
        else None
    ) or default_mime_type
```

(nit) feels more pythonic rather than initializing a variable and then populating it before checking if none